makefile.shade

```diff
@@ -688,128 +774,97 @@ var RC_FILES = '${FindAllFiles("Resource.rc", BOOTSTRAPPER_EXE_NAME, BOOTSTRAPPE
+    }
 
-        if (result != 0)
-        {
-            throw new InvalidOperationException(string.Format("Return value for the command listing assemblies for project under {0} is not zero ({1}).", projectFolder, result));
-        }
+macro name='KTest' projectFile='string'
+    k-test
+ 
 
-        try
+functions @{
+    private static readonly string _defaultLocalRuntimeHomeDir = ".dnx";
+    private static readonly string _runtimesSubDir = "runtimes";
+
+    bool CanBuildForLinux
+    {
+        get 
         {
-            return new HashSet<string>(File.ReadAllLines(tmpFile).Where(l => File.Exists(l)));
+            return string.Equals(Uname(), "Linux");
         }
-        catch (Exception ex)
+    }
+    
+    bool CanBuildForDarwin
+    {
+        get 
         {
-            throw new InvalidOperationException(string.Format("Unexpected exception thrown while parsing assemblies list for project under {0}. ({1})", projectFolder, ex.Message));
+            return string.Equals(Uname(), "Darwin");
         }
     }
 
-    string FindPackageDirectory(string name)
+    bool CanBuildForWindows
     {
-        var packagesDir = Environment.GetEnvironmentVariable("DNX_PACKAGES");
-
-        if (string.IsNullOrEmpty(packagesDir))
-        {
-            var profileDirectory = Environment.GetEnvironmentVariable("USERPROFILE");
-
-            if (string.IsNullOrEmpty(profileDirectory))
-            {
-                profileDirectory = Environment.GetEnvironmentVariable("HOME");
-            }
-
-            packagesDir = Path.Combine(profileDirectory, _defaultLocalRuntimeHomeDir, _packagesSubDir);
-        }
-        
-        var packageMetadataDir = Path.Combine(packagesDir, name);
-        
-        if (!Directory.Exists(packageMetadataDir))
+        get 
         {
-            return null;
+            var p = (int)Environment.OSVersion.Platform;
+            return (p != 4) && (p != 6) && (p != 128);
         }
-        
-        return Directory.GetDirectories(packageMetadataDir)
-                        .OrderByDescending(d => d)
-                        .FirstOrDefault();
     }
-    
-    bool IsMsbuildInstalled(string msBuildPath)
+
+    bool IsTravis 
     {
-        if (!File.Exists(msBuildPath))
+        get
         {
-            Log.Warn("msbuild version 14 not found. Please ensure you have the VS 2015 C++ SDK installed.");
-            return false;
+            var travis = Environment.GetEnvironmentVariable("TRAVIS");
+            return string.Equals(travis, "true");
         }
-    
-        return true;
     }
 
-    string[] FindAllProjects(params string[] folders)
-    {
-      return FindAllFiles("project.json", folders);
+    public string Uname()
+    {    
+        string uname;
+        ExecuteAndRedirectOutput("uname", "", out uname);
+        return string.IsNullOrEmpty(uname) ? null : uname.Trim();
     }
 
-    string[] FindAllFiles(string fileName, params string[] folders)
+    public void Clang(string args)
```

Comment:
In the future I think we should just search for clang/clang++-3.5 on the PATH and use what we find instead of scoping this to travis.  The -3.5 suffix is a Ubuntu-ism and an artifact of us bringing things up on Ubuntu 14.04.

If you file an issue against me I can fix this.